04_pig_latin/pig_latin.rb

```diff
@@ -1 +1,26 @@
 #write your code here
+def translate(string)
+    vowels = ["a", "e", "i", "o", "u"]
+    words = string.split(" ")
+    words = words.map do |word|#for each word in string
+        if vowels.include?(word[0])#first letter is vowel
+            word << "ay"
+        elsif word.start_with?("qu")
+            word[2, word.length] << word[0, 2] << "ay"
+        elsif vowels.include?(word[1])#second letter is vowel
+            word[1, word.length] << word[0] << "ay"
+        elsif word.slice(1, word.length).start_with?("qu")
+            word[3, word.length] << word[0, 3] << "ay"
+        elsif vowels.include?(word[2])#third letter is vowel
+            word[2, word.length] << word[0, 2] << "ay"
+        elsif vowels.include?(word[3])#fourth letter  not vowel
+            word[3, word.length] << word[0, 3] << "ay"
+        end#vowel check
+    end#for each word in string
+    words.join (" ")
+end#translate
```

Comment:
I really like this solution! 💯  Looking into this a bit, I learned today that the shovel operator on strings is slightly more performant than concatenation. Very cool!